qutebrowser/browser/commands.py

```diff
@@ -110,20 +110,25 @@ def _current_widget(self):
         return widget
 
     def _open(self, url, tab=False, background=False, window=False,
-              explicit=True):
+              explicit=True, private=None):
         """Helper function to open a page.
 
         Args:
             url: The URL to open as QUrl.
             tab: Whether to open in a new tab.
             background: Whether to open in the background.
             window: Whether to open in a new window
+            private: If opening a new window, open it in private browsing mode.
+                     If not given, inherit the current window's mode.
         """
         urlutils.raise_cmdexc_if_invalid(url)
         tabbed_browser = self._tabbed_browser
         cmdutils.check_exclusive((tab, background, window), 'tbw')
-        if window:
-            tabbed_browser = self._new_tabbed_browser()
+        if private is None:
+            private = self._tabbed_browser.private
+
+        if window or private:
```

Comment:
So `--private` implies `--window`, but doesn't get the exclusiveness check. `:open -p -t` should display an error just like `:open -w -t`.